### PR TITLE
Feat 76 Configure development environment / proxy API calls to backend

### PR DESCRIPTION
**Pull request by jhackshaw-dds**
Thanks for submitting a pull request! Below are a few things you can do to help us more quickly review your changes.

## Checklist
I have…
- [x] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [x] thoroughly outlined below the steps necessary to test my changes.
- [x] review and update SSAG documentation if needed.
- [x] attached screenshots illustrating relevant behavior before and after my changes.

## Summary of Changes
This pull request…
- Add docker-compose configuration for development environment with hot reloading
- Dynamically generate NGINX configuration based on environment variables
- Create/test RootAPI connection class for frontend API calls
- Add cors header support to django for API calls from different domain
- Configure allowed cors hosts dynamically via environment
- Add a "API Test" button that makes an API call for validating that the front and backend can communicate
- Update CONTRIBUTING.md with instructions for starting development environment

## Testing
To verify the changes proposed in this pull request…
1. Start local development environment via the following instructions taken from the CONTRIBUTING.md instructions:

- A docker-compose.yml file is included in the repository that quickly and easily allows the execution of the entire system for rapid local development. (all commands are specified from the `deploy/compose` directory):
  1. Copy the example environment file to a new file called `.env`
      -  `cp example.env .env`
  2. Edit the `.env` file and create a username and password for the database connection, and set your secret key to be a random value.
      - Note these values will only apply to your local environment
  3. Execute `docker-compose up --build`
  4. After the build process, navigate to `http://localhost:3000` in a browswer
      - Note that hot reloading is enabled for changes to front and backend source files

## Screenshots
Screen Shot 2020-02-04 at 4 22 24 PM

### CODE REVIEW

**Comment by SCasarottoDDS** [comment]
Is this section intended for the reviewer to know how to test or to show how you tested that things work as expected? I see whats in the `CONTRIBUTING.md` so that might be fine.

> Testing
> To verify the changes proposed in this pull request…
> 
> Spin up local development infrastructure via docker-compose in the deploy/compose folder per the instructions in CONTRIBUTING.md
> Run unit tests

Would it be too much trouble to list the exact things to run to spin up and test? Is it just a list of commands?

File: backend/app/settings.py [hunk 1]
```diff
@@ -70,49 +45,59 @@
...
 
 # Database
 # https://docs.djangoproject.com/en/3.0/ref/settings/#databases
-
 DATABASES = {
     "default": {
-        "ENGINE": "django.db.backends.sqlite3",
-        "NAME": os.path.join(BASE_DIR, "db.sqlite3"),
+        "ENGINE": "django.db.backends.postgresql",
+        "HOST": os.environ.get("POSTGRES_HOST", "localhost"),
```
Comment (jordangov) [inline backend/app/settings.py]: let's be consistent, up above we use named params, so let's update these to:
`os.environ.get("POSTGRES_HOST", default="localhost")`
(and update the others)

File: frontend/nginx/nginx.http.template [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header Host $http_host;
+        proxy_redirect off;
+        proxy_pass http://backend_proxy;
+    }
+}
+
+ 
+### AUTHENTICATION ##
```
Comment (jordangov) [inline frontend/nginx/nginx.http.template]: That's a lotta commented out code we got there... do we need it? Why not have this in a branch or git stash? We shouldn't be committing commented out code.

File: frontend/nginx/nginx.https.template [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    location @app {
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header Host $http_host;
+        proxy_redirect off;
+        proxy_pass http://backend_proxy;
+    }
+}
+
+### AUTHENTICATION ##
```
Comment (jordangov) [inline frontend/nginx/nginx.https.template]: More commented out code... :P

File: frontend/src/api/auth.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import root from "./root";
+
+export default class AuthApi {
+  public static login() {
+    return root.get("/logs/");
```
Comment (jordangov) [inline frontend/src/api/auth.ts:6]: Hmm... assuming this is a placeholder? If so, let's comment to that effect in the file so people aren't guessing about what "logs" this is showing.

File: frontend/nginx/nginx.https.template (hunk 3 above)
Comment (jhackshaw-dds) [reply frontend/nginx/nginx.https.template]: Agreed! I'll save this section until we are ready for authentication

File: frontend/src/api/auth.ts (hunk 4 above)
Comment (jhackshaw-dds) [reply frontend/src/api/auth.ts:6]: Yes, it's a placeholder. I will make that more clear.

File: backend/app/settings.py (hunk 1 above)
Comment (jhackshaw-dds) [reply backend/app/settings.py]: Updated

**Review by dds-hansolo — APPROVED** [review]
Testing was completed by Stuart and myself verifying both the https and http builds of the application are working locally, and the setup directions for other developers are complete.

**Pull request merged**